core/src/test/java/org/apache/iceberg/TestUpdatePartitionSpec.java

```diff
@@ -614,6 +605,65 @@ public void testRemoveAndAddMultiTimes() {
     V2Assert.assertEquals("Should match expected spec", v2Expected, updated);
   }
 
+  @Test
+  public void testRemoveAndUpdateWithIdentity() {
+    PartitionSpec expected = PartitionSpec.builderFor(SCHEMA)
+            .identity("ts")
+            .build();
+    PartitionSpec updated = new BaseUpdatePartitionSpec(formatVersion, expected)
+            .removeField("ts")
+            .addField("ts")
+            .apply();
+
+    if (formatVersion == 1) {
+      Assert.assertEquals("Should match expected spec field size", 2, updated.fields().size());
+      Assert.assertEquals("Should match expected field name", "ts",
+              updated.fields().get(0).name());
+      Assert.assertEquals("Should match expected field name", "ts",
+              updated.fields().get(1).name());
```

Comment:
I think that the reason this tries to modify the name check is this test, where the field is replaced with a void transform in v1. Instead, I think this should rename the void partition field like we do in other cases. You can rename the old field to `ts_1000` using the partition field ID to make the name unique. That's a better solution than allowing conflicting names.

Reply:
thanks for the review @rdblue ! I tried to rename the field, however the test was failed due to the limitation that we cannot delete and rename a partition col (https://github.com/apache/iceberg/blob/619603c443a200f687284c01d13e5961a920c036/core/src/main/java/org/apache/iceberg/BaseUpdatePartitionSpec.java#L221-L222), 

so I end up adding a hashset in partition set to store the fields being transformed with Void, 

please let me know if it sounds good to you, thanks!

Reply:
You should just need to add a way to bypass that check since this isn't a rename that is supplied by the user. Maybe just allow a rename if it's a void field.

Also, did you push your changes? It looks like this still modifies `PartitionSpec`.

Reply:
ah, sure, if it is ok, then it is simpler, just pushed the changes